src/NServiceBus.Core.Tests/Serializers/MessageDeserializerResolverTests.cs

```diff
@@ -54,5 +54,20 @@ public void NoContentTypeFallsBackToDefaultSerialization()
             var serializer = resolver.Resolve(headers);
             Assert.IsInstanceOf<XmlMessageSerializer>(serializer);
         }
+
+        [Test]
+        public void MultipleDeserializersWithSameContentTypeShouldThrowException()
+        {
+            var mapper = new MessageMapper();
+            var xml = new XmlMessageSerializer(mapper, new Conventions());
+            var json1 = new JsonMessageSerializer(mapper);
+            var json2 = new JsonMessageSerializer(mapper);
+
+            Assert.That(() => new MessageDeserializerResolver(xml, new IMessageSerializer[]
```

Comment:
You could have used the FakeSerializer which allows specifying a content type instead of the real ones.